### DISCUSSION TITLE
Docker always expose port 3000

### DISCUSSION THREAD

**Discussion by swuecho** [PostgREST/postgrest#1904] [Q&A] [answered]
### Environment

* PostgreSQL version: (if using docker, specify the image)
* PostgREST version: (if using docker, specify the image)
* Operating system:

### Description of issue

I have two instances of postgrest. one of them do not use the 3000 port, but the port is still exposed. 
```
pgrest_1     /bin/postgrest                   Up      0.0.0.0:3000->3000/tcp
pgrest_2    /bin/postgrest                   Up      3000/tcp, 0.0.0.0:3001->3001/tcp
```

**Comment by swuecho** [comment]
https://github.com/PostgREST/postgrest/blob/7ca0d469367d480804b2b8b91bdceb653ada754a/nix/tools/docker/README.md

`3000` is hardcoded to the Dockerfile.  Delete the  line `expose 3000` ?

**Comment by colearendt** [comment, marked as answer]
How are you running your containers? Why not build your `pgrest_2` image using port 3000 too, and then port-map when you _run_ the container? This is the usual pattern I have seen. Gets around the port collision neatly while allowing your containers to be built consistently 😄 
```
docker run -it --rm -p 3001:3000 --name pgrest_2 postgrest/postgrest
```

**Reply by swuecho** [reply]
nice!
